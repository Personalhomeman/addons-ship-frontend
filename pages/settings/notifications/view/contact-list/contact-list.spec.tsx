import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Toggle } from '@bitrise/bitkit';

import ContactList, { Props } from '.';

describe('NotificationSettings View ContactList', () => {
  const defaultProps: Props = {
    appContacts: [
      {
        email: 'bit.bot@bitrise.io',
        isConfirmed: true,
        notificationPreferences: {
          newVersion: true,
          successfulPublish: true,
          failedPublish: true
        }
      },
      {
        email: 'purr.request@bitrise.io',
        isConfirmed: false,
        notificationPreferences: {
          newVersion: false,
          successfulPublish: false,
          failedPublish: true
        }
      }
    ],
    onNotificationPreferenceChanged: jest.fn()
  };

  it('renders correctly', () => {
    const tree = shallowToJson(shallow(<ContactList {...defaultProps} />));

    expect(tree).toMatchSnapshot();
  });

  describe('onNotificationPreferenceChanged', () => {
    const preferenceKeys = ['newVersion', 'successfulPublish', 'failedPublish'];
    let wrapper: ShallowWrapper;
    const onNotificationPreferenceChanged = jest.fn();
    const appContacts = [
      {
        email: 'purr.request@bitrise.io',
        isConfirmed: false,
        notificationPreferences: {
          newVersion: false,
          successfulPublish: false,
          failedPublish: true
        }
      }
    ];

    beforeEach(() => {
      wrapper = shallow(
        <ContactList appContacts={appContacts} onNotificationPreferenceChanged={onNotificationPreferenceChanged} />
      );
      onNotificationPreferenceChanged.mockReset();
    });

    preferenceKeys.forEach((key, idx) => {
      it(`triggers onNotificationPreferenceChanged correctly for ${key}`, () => {
        const current = appContacts[0].notificationPreferences[key];
        wrapper
          .find(Toggle)
          .at(idx)
          .simulate('change');

        expect(onNotificationPreferenceChanged).toHaveBeenCalledWith(appContacts[0].email, key, !current);
      });
    });
  });
});
