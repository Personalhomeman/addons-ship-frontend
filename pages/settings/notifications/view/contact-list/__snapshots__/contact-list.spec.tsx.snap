// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettings View ContactList renders correctly 1`] = `
<Table
  type="flat"
>
  <TableHeader>
    <TableHeaderRow>
      <TableHeaderCell>
        Email
      </TableHeaderCell>
      <TableHeaderCell />
      <TableHeaderCell
        align="middle"
      >
        New
        <br />
        Version
      </TableHeaderCell>
      <TableHeaderCell
        align="middle"
      >
        Successful
        <br />
        Publish
      </TableHeaderCell>
      <TableHeaderCell
        align="middle"
      >
        Failed
        <br />
        Publish
      </TableHeaderCell>
      <TableHeaderCell />
    </TableHeaderRow>
  </TableHeader>
  <TableBody>
    <TableRow
      className="roww"
      key="bit.bot@bitrise.io"
    >
      <TableCell
        shrink={true}
      >
        bit.bot@bitrise.io
      </TableCell>
      <TableCell
        align="middle"
      >
        Confirmed
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Toggle
          alignChildren="middle"
          checked={true}
          onChange={[Function]}
        />
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Toggle
          alignChildren="middle"
          checked={true}
          onChange={[Function]}
        />
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Toggle
          alignChildren="middle"
          checked={true}
          onChange={[Function]}
        />
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Base
          borderRadius="x1"
          className="delete"
          padding="x1"
        >
          <Icon
            color="red-3"
            name="Trash"
            size="1.5rem"
          />
        </Base>
      </TableCell>
    </TableRow>
    <TableRow
      className="roww"
      key="purr.request@bitrise.io"
    >
      <TableCell
        shrink={true}
      >
        purr.request@bitrise.io
      </TableCell>
      <TableCell
        align="middle"
      >
        Pending
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Toggle
          alignChildren="middle"
          checked={false}
          onChange={[Function]}
        />
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Toggle
          alignChildren="middle"
          checked={false}
          onChange={[Function]}
        />
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Toggle
          alignChildren="middle"
          checked={true}
          onChange={[Function]}
        />
      </TableCell>
      <TableCell
        shrink={true}
      >
        <Base
          borderRadius="x1"
          className="delete"
          padding="x1"
        >
          <Icon
            color="red-3"
            name="Trash"
            size="1.5rem"
          />
        </Base>
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
